escrow/payment_storage.go

```diff
@@ -12,13 +13,11 @@ type PaymentStorage struct {
 
 // NewPaymentStorage returns new instance of PaymentStorage
 // implementation
-func NewPaymentStorage(atomicStorage AtomicStorage) *PaymentStorage {
+func NewPaymentStorage(atomicStorage AtomicStorage, metadata *blockchain.ServiceMetadata) *PaymentStorage {
 	return &PaymentStorage{
 		delegate: &TypedAtomicStorageImpl{
-			atomicStorage: &PrefixedAtomicStorage{
-				delegate:  atomicStorage,
-				keyPrefix: "/payment/storage",
-			},
+
+			atomicStorage:     NewPrefixedAtomicStorage(atomicStorage, "/payment/storage", metadata.MpeAddress),
```

Comment:
What is the reason to prefix PaymentStorage by MPE contract address? MPE contract address can be changed even if network is same and we should keep payments for both MPE contracts. For example such thing may happen when MPE contract is updated.

Reply:
having MPE contract ensures the payments from both these contracts are still stored seprately no ? else the payments can get corrupted with old and new , specially when the key is same no ?

Reply:
Ok, you are right, I thought that MPE contract address somehow is taken into account in PaymentID but I see it is not true. Let's summarize it (please correct me if I wrong):
- we have three storages: PaymentChannelStorage, PaymentStorage and LockStorage;
- both PaymentChannelStorage and LockStorage are actual for current MPE contract only;
- PaymentStorage in fact can be used with different versions of the MPE contract, for example if MPE contract was upgraded but old payments need to be saved to be claimed;
- we can also keep payments for each MPE separately then to get old payments some daemon instance should be switched temporary for old MPE contract.

So I would suggest add MPE contract address + network_id as environment_id as I explained https://github.com/singnet/snet-daemon/pull/442#issuecomment-579653252 and remove it from all other places.

Reply:
Thanks @vsbogd , say we have the below scenario 
On MPE contract 1 , 
lets say the LockStorage , the key is on the channel, say channel id 2 , now for some reason , this gets marked locked in the Daemon for MPE contract 1 , ( etcd storage is same ) 

On MPE contract 2 
Lets say a brand new request comes again on a channel say , channe id 2 ( brand new channel) , if we dont distinguish the locking storageof MPE1 and MPE2, then  the old lock from channel 2 on MPE1 will block the request from on Channel 2 from the new MPE contract .no ?

Reply:
Well, it is not I was going to say. I will try to rephrase the solution suggested taking https://github.com/singnet/snet-daemon/pull/443#discussion_r373450556 into account. We have base AtomicStorage in `components.go` which is used to build other storages: PaymentChannelStorage, PaymentStorage, LockStorage for channels and new one LockStorage for free payments. What I am suggesting:
- get network id on daemon start;
- create new PrefixedStorage using `/<network_id>` as a prefix, use this storage as base for other storages (i.e. return it from `GetAtomicStorage` of `components.go`); this guarantees that storages for different networks never intersect;
- add new component MPESpecificStorage; it is also instance of PrefixedStorage using `/<mpe_contract_address>` as a prefix; as it is also based on storage from previous item the effective prefix is `/<network_id>/<mpe_contract_address>`; this guarantees that storages which are specific for MPE contract version don't intersect;
- use MPESpecificStorage as base for PaymentChannelStorage, PaymentStorage, LockStorage for channels;
- use GetAtomicStorage as base for LockStorage for free payments.

As result the following hierarchy of storages should be built:
```
AtomicStorage(no prefix)
  PrefixedStorage(/<network_id>) returned by `components.GetAtomicStorage()`
    LockStorage for free payments
    PrefixedStorage(/<mpe_contract_address>) - returned by `components.GetMpeSpecificStorage()`
      PaymentChannelStorage
      PaymentStorage
      LockStorage for channels
```